src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.Windows.cs

```diff
@@ -68,5 +68,47 @@ void Cleanup(string username, string _)
                 }
             }
         }
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [InlineData(ProcessWindowStyle.Normal)]
+        [InlineData(ProcessWindowStyle.Hidden)]
+        [InlineData(ProcessWindowStyle.Minimized)]
+        [InlineData(ProcessWindowStyle.Maximized)]
+        public void TestWindowStyleUseShellExecuteFalse(ProcessWindowStyle windowStyle)
+        {
+            // "x y" where x is the expected dwFlags & 0x1 result and y is the wShowWindow value
+            string procArg = windowStyle switch
+            {
+                ProcessWindowStyle.Hidden => "1 0",
+                ProcessWindowStyle.Minimized => "1 2",
+                ProcessWindowStyle.Maximized => "1 3",
+                _ => "0 0",
+            };
+
+            Process p = CreateProcess((string procArg) =>
+            {
+                Interop.GetStartupInfoW(out var si);
+
+                string[] argSplit = procArg.Split(" ");
+                int expectedFlags = int.Parse(argSplit[0]);
+                short expectedWindow = short.Parse(argSplit[1]);
+
+                if ((si.dwFlags & 0x00000001) == expectedFlags && si.wShowWindow == expectedWindow)
+                {
+                    return RemoteExecutor.SuccessExitCode;
+                }
+                else
+                {
+                    return -1;
+                }
```

Comment:
I think RemoteExecutor should be able to handle assertion failures. 
```suggestion
                Assert.Equal(expectedDwFlag, si.dwFlags);
                Assert.Equal(expectedWindowFlag, si.wShowWindow);
                
                return RemoteExecutor.SuccessExitCode;
```

Reply:
Awesome, I was not sure how advanced the `RemoteExecutor` was.